### DISCUSSION TITLE
Log4j jar that comes with mirth connect install (2.0-2.14) BAD.

### DISCUSSION THREAD

**Discussion by Wet-Bandit** [nextgenhealthcare/connect#4911] [Q&A] [answered]
There is a vulnerability in the Java Log4j library.  I hope mirth does not update it to one of these bad versions.  If one is using any version between 2.0 and 2.14, they will need to upgrade to 2.15.0.   My customer wanted me to forward the information to Mirth Corp.  I removed log4j from my other web applications years ago.  That library has become bloatware and unnecessary.  There is regular java logging and also commons-logging to use. 

TLP: WHITE

MS-ISAC CYBERSECURITY ADVISORY

 

MS-ISAC ADVISORY NUMBER:

2021-158

 

DATE(S) ISSUED:

12/10/2021

 

SUBJECT:

A Vulnerability in Apache Log4j Could Allow for Arbitrary Code Execution

 

OVERVIEW:

A vulnerability has been discovered in Apache Log4j, a very ubiquitous logging package for Java. Successful exploitation of this vulnerability could allow for arbitrary code execution within the context of the systems and services that use the Java logging library, including many services and applications written in Java. Depending on the privileges associated with these systems and services, an attacker could then install programs; view, change, or delete data; or create new accounts with full user rights. If these systems and services have been configured to have fewer user rights, exploitation of this vulnerability could have less impact than if they were configured with administrative rights.

 

THREAT INTELLIGENCE:

According to numerous open source reports, Log4j is used with Apache software like Apache Struts, Solr, Druid, along with other technologies. Many websites of manufacturers and providers have been found to be affected including Apple, Twitter, Steam, Tesla and more. Threat actors will likely include payloads in simple HTTP connections, either in a User-Agent header or trivial POST form data. In addition, it has been reported that organizations are already seeing signs of exploitation in the wild with further attempts on other websites likely.

 

SYSTEMS AFFECTED:

Apache Log4j between versions 2.0 and 2.14.1
 

RISK:

Government:

Large and medium government entities: High
Small government entities: High
Businesses:

Large and medium business entities: High
Small business entities: High
Home users: High

 

TECHNICAL SUMMARY:

A vulnerability has been discovered in Apache Log4j, a very ubiquitous logging package for Java. This vulnerability resides in the JNDI lookup feature of the log4j library. The JNDI lookup feature of log4j allows variables to be retrieved via JNDI - Java Naming and Directory Interface. This is an API that provides naming and directory functionality to Java applications. While there are many possibilities, the log4j API supports LDAP and RMI (Remote Method Invocation).

 

Successful exploitation of this vulnerability could allow for arbitrary code execution within the context of the systems and services that use the Java logging library, including many services and applications written in Java. Depending on the privileges associated with these systems and services, an attacker could then install programs; view, change, or delete data; or create new accounts with full user rights. If these systems and services have been configured to have fewer user rights, exploitation of this vulnerability could have less impact than if they were configured with administrative rights.

 

RECOMMENDATIONS:

We recommend the following actions be taken:

·         Apply the latest patches (version 2.15.0) provided by Apache after appropriate testing.

·         Run all systems and services as a non-privileged user (one without administrative privileges) to diminish the effects of a successful attack.

·         Apply the Principle of Least Privilege to all systems and services.

 

REFERENCES:

 

CVE:

https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-44228

 

SANS Technology Institute:

https://isc.sans.edu/diary/28120

 

ZDNet:

https://www.zdnet.com/article/security-warning-new-zero-day-in-the-log4j-java-library-is-already-being-exploited/

 

Ars Technica:

https://arstechnica.com/information-technology/2021/12/minecraft-and-other-apps-face-serious-threat-from-new-code-execution-bug/

 

 

Multi-State Information Sharing and Analysis Center (MS-ISAC)

Elections Infrastructure Information Sharing and Analysis Center (EI-ISAC)

31 Tech Valley Drive

East Greenbush, NY 12061

**Comment by pacmano1** [comment]
There are several threads already discussing this.    Please read those threads and issues already posted.  Your post isn't really all that insightful.   If you read those you would see that 1) you are late to the party 2) Mirth does not have log4j v2, and NG has already responded about v1 and v2 plans.

**Reply by tonygermano** [reply]
See https://github.com/nextgenhealthcare/connect/discussions/4892

**Comment by Wet-Bandit** [comment, marked as answer]
I added more info above.  The library in mirth connect is currently log4j-1.2.16.jar, so we are good just leaving this old one in place.

**Comment by JonEBee** [comment]
Mirth 3.x still uses log4j 1.2 and also does not use the log4j ServerSocket call, so Mirth engines are safe for now.
